realm-jni/src/io_realm_internal_SharedGroup.cpp

```diff
@@ -150,21 +150,21 @@ JNIEXPORT jlong JNICALL Java_io_realm_internal_SharedGroup_nativeBeginImplicit
 }
 
 JNIEXPORT void JNICALL Java_io_realm_internal_SharedGroup_nativeAdvanceRead
-(JNIEnv *env, jobject, jlong native_ptr)
+(JNIEnv *env, jobject, jlong native_ptr, jlong native_replication_ptr)
```

Comment:
`ClientHistory` is inherited from Replication. And in core's `SharedGroup` constructor, `Replication` is still used as `SharedGroup(Replication& repl, DurabilityLevel durability = durability_Full, const char* encryption_key = 0)` . I think it is better to keep the name `replication` for now to avoid it becoming more confusing.